src/lib/components/tablev2/SingleSelectableContent.tsx

```diff
@@ -140,6 +143,35 @@ export function SingleSelectableContent({
     handleScrollbarWidth,
   } = useTableScrollbar();
 
+  const headerRef = useRef<HTMLDivElement | null>(null);
+  const bodyRef = useRef<FixedSizeList<Row<DATA_ROW>[]> | null>(null);
+  const currentHeadRef = !!headerRef.current;
+  const currentBodyRef = !!bodyRef.current;
+  useEffect(() => {
+    if (bodyRef.current && headerRef.current) {
+      const listener = (event: Event) => {
+        headerRef.current.scrollTo({
+          left: (event.target as HTMLDivElement).scrollLeft,
+          top: 0,
+        });
+      };
+
+      /*
+      We intentionally use _outerRef prop here despite the fact that it is 
+      internal use only and not typed, as it is the only way for us to access to the scrollable element
+      */
+      //@ts-expect-error
+      (bodyRef.current._outerRef as HTMLDivElement).addEventListener(
+        'scroll',
+        listener,
+      );
+      return () => {
+        //@ts-expect-error
+        bodyRef.current._outerRef.removeEventListener('scroll', listener);
+      };
+    }
+  }, [currentHeadRef, currentBodyRef]);
```

Comment:
This code is duplicated between SignleSelectableContent and MultiSelectableContent. I'm sure we can do better ;) maybe extract the logic to a custom hook shared in TableCommon ?